### ISSUE TITLE
MSVC build is broken

### ISSUE THREAD

**Issue by db4** [ocaml-omake/omake#28]
1. In commit 231b2cc one added .opt targets to mk/make_gen but used the wrong threads library ($$(THREADSLIB) instead of correct $$(THREADSLIB_OPT)
2.  In commit 5852c7d one removed unixsupport.h from ALLFILES_clib that was actually required. What was the purpose of doing that? Did the committer ever tested his modifications?

Why case 1 is easy to fix, case 2 reveals another problem. Gerd, it's a pity that you accepted all those poorly commented arbitrary commits from bobzhang1988@gmail.com into your "official" repository without testing (and maybe rewriting). It's quite hard to understand now why and when the build system was broken w.r.t. the latest original omake 0.9.8.6

**Comment by gerdstolpmann** [comment]
1. is fixed.

2. should not be an issue: unixsupport.h is now installed with OCaml - it used not to be in ancient versions, and that is the only reason why a copy was in the tree. If there is any problem still, then it's something else. What's the issue precisely?

Regarding Bob's patches, these were essentially the starting point for my work. I roughly scrolled through them, and they are mostly reorganizations and language modernizations, i.e. things we want to have. I thought any remaining problem would have to be ironed out, but that it was better to have these patches.

Regarding Windows in general, I don't have a setup for MSVC. My testing is limited to mingw (and I made that working). Any help in this direction is highly welcome. Generally, the development direction is to unify with Unix as much as possible, in particular in the bootstrap. I have now an OCaml implementation of "make" (see branch "homemade"), so the days of nmake are counted. Of course, this will introduce new bugs, but that's the price to pay for not standing still.

But anyway, thanks for trying out the git version, and helping to track problems down.

**Comment by neolem** [comment]
@gerdstolpmann you can download msvc build tools freely at https://www.microsoft.com/en-us/download/details.aspx?id=48159

**Comment by gerdstolpmann** [comment]
Thanks neolem for the pointer. I knew that is was free. It is just the time I'd have to invest to make it really working.

**Comment by db4** [comment]
OK, thanks for the explanation.

Regarding  unixsupport.h - the problem is that it is not found during the bootstrap phase. omake used to set `-I ${OCAMLLIB}` for C compiler (not `${OCAMLLIB}/caml`) and for whatever reason had its own unixsupport.h.

I started my own attempts to make MSVC build alive again (https://github.com/db4/omake/commits/master) but right now it's still broken. I will post PR when it's done.

**Comment by gerdstolpmann** [comment]
So maybe it's a path issue then. I see that e.g. mlvalues.h is referenced as caml/mlvalues.h. Maybe doing the same for unixsupport.h solves the issue for you.

**Comment by gerdstolpmann** [comment]
FYI, tried to install ocaml with msvc but failed. The Windows 7 SDK mentioned in the README.win32 cannot be installed on Windows 10 (it wants .net 4 although it is already there). I tried something more recent instead (Windows 10 SDK, VS Express) but that resulted only in libraries not found; most funnily I got a

```** Fatal error: Cannot find file "OLDNAMES"```

from flexlink.

So, regarding the support of msvc in OCaml, I'd summarize that it doesn't work with the freely available tools, at least when you don't want to invest a lot of time into that.

Why can't the Windows folks never come up with builds that just work?
